src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/WindowsRuntime/Windows/UI/ViewManagement/UISettings.cs

```diff
@@ -0,0 +1,111 @@
+using MS.Internal.PresentationCore.WindowsRuntime;
+using System;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Windows;
+using System.Windows.Interop;
+using System.Windows.Media;
+
+namespace MS.Internal.WindowsRuntime
+{
+    namespace Windows.UI.ViewManagement
+    {
+        internal class UISettings
+        {
+            private static readonly bool _isSupported;
+            private static object _winRtInstance;
+            private UISettingsRCW.IUISettings3 _uisettings;
+
+            static UISettings()
+            {
+                try
+                {
+                    if (GetWinRTInstance(forceInitialization: true) == null)
+                    {
+                        _isSupported = false;
+                    }
+                    _isSupported = true;
+                }
+                catch
+                {
+                    _isSupported = false;
+                }
+            }
+
+            internal UISettings()
+            {
+                if (!_isSupported)
+                {
+                    throw new PlatformNotSupportedException();
+                }
+
+                try
+                {
+                    UISettingsRCW.IUISettings3 uisettings;
+
+                    try
+                    {
+                        uisettings = GetWinRTInstance() as UISettingsRCW.IUISettings3;
+                    }
+                    catch (COMException)
+                    {
+                        uisettings = GetWinRTInstance(forceInitialization: true) as UISettingsRCW.IUISettings3;
```

Comment:
Just a question: why is this worth retrying, can the previous instance get somehow outdated?

Reply:
I haven't added the comments here, but this pattern is followed in InputPane.cs, and there it was mentioned that the object can go stale, but that comment was regarding an activation factory instance.